Makefile

```diff
@@ -0,0 +1,39 @@
+# Makefile for running typical developer workflow actions.
+# To run actions in a subdirectory of the repo:
+#   make lint build dir=translate/snippets
+
+INTERFACE_ACTIONS="build test lint"
+
+.ONESHELL: #ease subdirectory work by using the same subshell for all commands
+.-PHONY: *
+
+# Default to current dir if not specified.
+dir ?= $(shell pwd)
+
+export GOOGLE_CLOUD_PROJECT = ${GOOGLE_SAMPLES_PROJECT}
+
+build:
```

Comment:
The commands after "cd" is executed at the root of the project and not in the directory specified. Applies to all actions.

Reply:
That's normally the case with a Makefile, but the [`.ONESHELL` option](https://www.gnu.org/software/make/manual/html_node/One-Shell.html) above ensures all commands are run in the same shell, so the `cd` does take effect for all commands.

Reply:
Yeah, initially I wasn't able to get the makefile run with the ONESHELL option. But turns out its my corrupted IDE. I verified it works with Cloud Shell. Thanks for clarifying.